--- VoodooPS2SynapticsTouchPad.cpp	2016-12-15 09:00:23.000000000 +0800
+++ /Users/iceman/Developer/OS-X-Voodoo-PS2-Controller/VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.cpp	2016-12-15 09:00:52.000000000 +0800
@@ -177,10 +177,6 @@
 	yrest=0;
     lastbuttons=0;
     
-    skippyThresh=0;
-    lastdx=0;
-    lastdy=0;
-    
     // intialize state for secondary packets/extendedwmode
     xrest2=0;
     yrest2=0;
@@ -559,7 +555,7 @@
     setProperty(kIOHIDScrollAccelerationTypeKey, kIOHIDTrackpadScrollAccelerationKey);
 	setProperty(kIOHIDScrollResolutionKey, _scrollresolution << 16, 32);
     // added for Sierra precise scrolling (credit usr-sse2)
-    setProperty("HIDScrollResolutionX", (_scrollresolution >> 2) << 16, 32);
+    setProperty("HIDScrollResolutionX", _scrollresolution << 16, 32);
     setProperty("HIDScrollResolutionY", _scrollresolution << 16, 32);
     
     //
@@ -1613,7 +1609,7 @@
                         inSwipeUp=1;
                         inSwipeDown=0;
                         ymoved = 0;
-                        //_device->dispatchKeyboardMessage(kPS2M_swipeUp, &now_abs);
+                        _device->dispatchKeyboardMessage(kPS2M_swipeUp, &now_abs);
                         break;
                     }
                     if (ymoved < -swipedy && !inSwipeDown)
@@ -1621,7 +1617,7 @@
                         inSwipeDown=1;
                         inSwipeUp=0;
                         ymoved = 0;
-                        //_device->dispatchKeyboardMessage(kPS2M_swipeDown, &now_abs);
+                        _device->dispatchKeyboardMessage(kPS2M_swipeDown, &now_abs);
                         break;
                     }
                     if (xmoved < -swipedx && !inSwipeRight)
@@ -1750,7 +1746,6 @@
         momentumscrollcurrent = 0;
     }
 
-    
     // switch modes, depending on input
 	if (touchmode==MODE_PREDRAG && isFingerTouch(z))
     {
@@ -1804,40 +1799,7 @@
 	if (touchmode==MODE_NOTOUCH && z>z_finger)
 		touchmode=MODE_MOVE;
     
-    // pointer jumpy fix;
-#if 1
-    if (touchmode == MODE_VSCROLL || touchmode == MODE_HSCROLL) {
-        
-    } else {
-    if (skippyThresh > 0) {
-        skippyThresh--;
-        if ((dx > 100 || dx < -100) && (dy > 100 || dy < -100)) {
-            dx = 0;
-            dy = 0;
-        }
-    }
-        if (lastdx == 0 && dx != 0) {
-            dx = 0;
-            lastdx = 1;
-        } else {
-            lastdx = dx;
-        }
-        if (lastdy == 0 && dy != 0) {
-            dy = 0;
-            lastdy = 1;
-        } else {
-            lastdy = dy;
-        }
-        if (dx == 0 && dy == 0) {
-            skippyThresh = 4;
-        } else {
-            skippyThresh--;
-        }
-    }
-#endif
-    
     // dispatch dx/dy and current button status
-    
     dispatchRelativePointerEventX(dx / divisorx, dy / divisory, buttons, now_abs);
     
     // always save last seen position for calculating deltas later
